### DISCUSSION TITLE
ScriptReference/Editor.DrawDefaultInspector

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#148] [Notes]
# ScriptReference/Editor.DrawDefaultInspector

https://docs.unity3d.com/ScriptReference/Editor.DrawDefaultInspector.html

**Comment by vertxxyz** [comment]
There is an undocumented method `DrawPropertiesExcluding(SerializedObject obj, params string[] propertyToExclude)` that will act like `DrawDefaultInspector` but you can specify serialized properties to not draw.  
For example you could provide `"m_Script"` to not draw the script field at the top of the component UI.

**Comment by vertxxyz** [comment]
For UI Toolkit editors you use [`CreateInspectorGUI`](https://docs.unity3d.com/ScriptReference/Editor.CreateInspectorGUI.html), and call [`InspectorElement.FillDefaultInspector`](https://docs.unity3d.com/ScriptReference/UIElements.InspectorElement.FillDefaultInspector.html) with your root and context objects, which will append the default inspector to the visual element. Note that the initialisation of the elements inside the root will be delayed until Bind is run, which is asynchronous.
